### ISSUE TITLE
`getHouse` action occasionally fails

### ISSUE THREAD

**Issue by ptrfrncsmrph** [labs12-real-estate/labs12-real-estate-FE#39]
### Expected Behavior
Complete the wizard form and see a newly created house associated with the user account in the `/overview` route.

### Current Behavior
The `getHouse` action (which is called when the `OverviewPage` mounts) occasionally fails, not sure if this has to do with AWS being temperamental or if there is some way that we're handling the `createHouse` mutation incorrectly (passing in the wrong type of input). I don't think it's the latter because we consistently get a successful response from AWS after running the `createHouse` mutation.

### Possible Solution
A band-aid solution is in place to use the data that we get back from `createHouse` as a fallback in case the `getHouse` action returns `undefined`. This works for populating the `OverviewPage` after the wizard form, but when that user logs out and back in, or refreshes the page, the data will have to rely on `getHouse` working correctly.

### Steps to Reproduce
- Fill out wizard form (I'm not sure if the bug relies on certain inputs)
- Complete the new user registration and submit the confirmation code
- When `getHouse` fails, in the console you would see the Redux logger middleware output:
```
{ type: "MAKE_HOUSE_SUCCESS", payload: { /* valid house data here */ } }
```
and then
```
{ type: "GET_HOUSE_SUCCESS", payload: undefined }
```
The `undefined` payload is what will throw the error if the page is refreshed or existing user comes to the `OverviewPage` from logging in instead of registering.

### Context (Environment)
User registration flow, after wizard form submit.

**Comment by ptrfrncsmrph** [comment]
After some digging, this issue is documented here: https://github.com/aws-amplify/amplify-cli/issues/352#issue-373996634.
*TL;DR* `getHouse` is calling the `listHouses` query which has an implicit `limit` parameter, and returning the first item in the array that matches the user ID. The array is truncated by the `limit`, so sometimes the matching house isn't in the first page.
So, possible solutions would be (in order of how much I would know how to implement):
- assign the `House` ID based off the current user ID and `getHouse` by ID instead of `listHouses`, made possible by this: https://github.com/aws-amplify/amplify-cli/pull/356. Drawback is that would throw an exception if you try to create a second house for a given user because of unique ID constraint so there could only be one house per user (even though that's how we're implementing the app anyways at this point, a bit restrictive).
- create a `User` _and_ a `House` on every registration, and use `@connection` directive to link them
- use a custom attribute assigned to the user (not in a users table, just in pool) to link them to their `House`
- Global Secondary Indices https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html
